src/EventListener/ProductTranslationUpdateListener.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Setono\SyliusRedirectPlugin\EventListener;
+
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Setono\SyliusRedirectPlugin\Resolver\InfiniteLoopResolverInterface;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Sylius\Component\Resource\Exception\UpdateHandlingException;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final class ProductTranslationUpdateListener
+{
+    private $request;
+    private $redirectionFactory;
+    private $redirectionRepository;
+    private $infiniteLoopResolver;
+    private $router;
+    private $validator;
+    private $flashBag;
+
+    public function __construct(RequestStack $requestStack,
+                                FactoryInterface $redirectionFactory,
+                                RepositoryInterface $redirectionRepository,
+                                InfiniteLoopResolverInterface $infiniteLoopResolver,
+                                RouterInterface $router,
+                                ValidatorInterface $validator,
+                                FlashBagInterface $flashBag
+    )
+    {
+        $this->request = $requestStack->getCurrentRequest();
+        $this->redirectionFactory = $redirectionFactory;
+        $this->redirectionRepository = $redirectionRepository;
+        $this->infiniteLoopResolver = $infiniteLoopResolver;
+        $this->router = $router;
+        $this->validator = $validator;
+        $this->flashBag = $flashBag;
+    }
+
+    public function postUpdate(LifecycleEventArgs $event): void
+    {
+        $productTranslation = $event->getObject();
+        if (!$productTranslation instanceof ProductTranslationInterface) {
+            return;
+        }
+
+        $uow = $event->getEntityManager()->getUnitOfWork();
+        $changeSet = $uow->getEntityChangeSet($productTranslation);
+        $this->handleAutomaticRedirectionCreation($productTranslation, $changeSet);
+    }
+
+    private function handleAutomaticRedirectionCreation(ProductTranslationInterface $productTranslation, $changeSet): void
```

Comment:
What's the reason for creating this method? Why not put the contents directly in `postUpdate`?

Reply:
Re usability if we need to listen to another event. Is it unecessary ?